src/picongpu/include/particles/ionization/byField/BSI/BSI.hpp

```diff
@@ -54,8 +62,62 @@ namespace ionization
             /* alias for electron species to be created */
             PMACC_AUTO(electrons,pst[typename MakeIdentifier<T_DestSpecies>::type()]);
             /* call the ionization function */
-            src.ionize(electrons, currentStep);
+
+            typedef T_SrcSpecies SrcSpeciesType;
+            typedef typename SrcSpeciesType::FrameType FrameType;
+            
+            DataConnector &dc = Environment<>::get().DataConnector();
+
+            fieldE = &(dc.getData<FieldE > (FieldE::getName(), true));
+            fieldB = &(dc.getData<FieldB > (FieldB::getName(), true));
+            
+            typedef typename PMacc::traits::Resolve<
+                typename GetFlagType<FrameType,interpolation<> >::type
+            >::type InterpolationScheme;
+            
+            /* margins around the supercell for the interpolation of the field on the cells */
+            typedef typename GetMargin<InterpolationScheme>::LowerMargin LowerMargin;
+            typedef typename GetMargin<InterpolationScheme>::UpperMargin UpperMargin;
+            
+            /* define type of frame ionizer 
+             * that moves over the frames with the particles and executes an operation on them */
+            typedef particles::ionization::IonizeParticlesPerFrame<IonizationAlgorithm, 
+                MappingDesc::SuperCellSize,
+                InterpolationScheme,
+                fieldSolver::NumericalCellType > FrameIonizer;
+            
+            /* relevant area of a block */
+            typedef SuperCellDescription<
+                typename MappingDesc::SuperCellSize,
+                LowerMargin,
+                UpperMargin
+                > BlockArea;
+            
+            /* 3-dim vector : number of threads to be started in every dimension */
+            dim3 block( MappingDesc::SuperCellSize::toRT().toDim3() );
+            
+            MappingDesc cellDescription;
```

Comment:
@psychocoderHPC default constructor of `src/libPMacc/include/mappings/kernel/MappingDescription.hpp` is used here:

```c++
    MappingDescription(DataSpace<DIM> localGridCells = DataSpace<DIM> (),
                       int borderSuperCells = 0,
                       int guardingSuperCells = 0) :
/* ... */
```

default `DataSpace<DIM> ()` is a `MappingDesc's DIM` vector with `0`'s.

We should add an assert to `__picKernelArea` in `./src/picongpu/include/simulation_classTypes.hpp` to avoid that *hard-to-interpret* CUDA internal error when called with zero blocks and zero block sizes.